tools/ports/sdl3.py

```diff
@@ -0,0 +1,130 @@
+  # get the port
+  ports.fetch_project('sdl3', f'https://github.com/libsdl-org/SDL/archive/{TAG}.zip', sha512hash=HASH)
+
+  def create(final):
+    diagnostics.warning('experimental', 'sdl3 port is still experimental')
+    root_dir = ports.get_dir('sdl3', SUBDIR)
+
+    # copy includes to a location so they can be used as 'SDL3/'
+    source_include_path = os.path.join(root_dir, 'include', 'SDL3')
+    ports.install_headers(source_include_path, target='SDL3')
+
+    # In addtion copy our pregenerated SDL_build_config.h file.
+    here = os.path.dirname(__file__)
+    ports.install_headers(os.path.join(here, 'sdl3'), 'SDL_build_config.h', target='SDL3')
+
+    glob_patterns = [
+      # Generic sources (from SDL3's CMakeLists.txt)
+      '*.c',
+      'atomic/*.c',
+      'audio/*.c',
+      'camera/*.c',
+      'core/*.c',
+      'cpuinfo/*.c',
+      'dynapi/*.c',
+      'events/*.c',
+      'io/*.c',
+      'io/generic/*.c',
+      'filesystem/*.c',
+      'gpu/*.c',
+      'joystick/*.c',
+      'haptic/*.c',
+      'hidapi/*.c',
+      'locale/*.c',
+      'main/*.c',
+      'misc/*.c',
+      'power/*.c',
+      'render/*.c',
+      'render/*/*.c',
+      'sensor/*.c',
+      'stdlib/*.c',
+      'storage/*.c',
+      'thread/*.c',
+      'time/*.c',
+      'timer/*.c',
+      'video/*.c',
+      'video/yuv2rgb/*.c',
+      'tray/*.c',
+      # Platform speecifc sources
+      'storage/generic/*.c',
+      'tray/unix/*.c',
+      'time/unix/*.c',
+      'timer/unix/*.c',
+      'main/emscripten/*.c',
+      'filesystem/posix/*.c',
+      'filesystem/emscripten/*.c',
+      'locale/emscripten/*.c',
+      'camera/emscripten/*.c',
+      'joystick/emscripten/*.c',
+      'joystick/virtual/*.c',
+      'power/emscripten/*.c',
+      'misc/emscripten/*.c',
+      'audio/emscripten/*.c',
+      'video/emscripten/*.c',
+      'video/offscreen/*.c',
+      'audio/disk/*.c',
+      'loadso/dlopen/*.c',
+      # Dummy backends, do we need to support these?
+      'camera/dummy/*.c',
+      'audio/dummy/*.c',
+      'video/dummy/*.c',
+      'haptic/dummy/*.c',
+      'sensor/dummy/*.c',
+    ]
+
+    flags = ['-sUSE_SDL=0']
+    if settings.PTHREADS:
+      glob_patterns.append('thread/pthread/*.c')
+      flags += ['-pthread']
+    else:
+      glob_patterns.append('thread/generic/*.c')
+
+    srcs = []
+    for pattern in glob_patterns:
+      matches = glob.glob(os.path.join(root_dir, 'src', pattern))
+      assert matches
+      srcs += matches
+
+    srcs = [os.path.join(root_dir, 'src', s) for s in srcs]
+    includes = [ports.get_include_dir('SDL3'), os.path.join(root_dir, 'src')]
+    ports.build_port(root_dir, final, 'sdl3', srcs=srcs, includes=includes, flags=flags)
+
+  return [shared.cache.get_lib(get_lib_name(settings), create, what='port')]
+
+
+def clear(ports, settings, shared):
+  shared.cache.erase_lib(get_lib_name(settings))
+
+
+def process_args(ports):
+  return ['-isystem', ports.get_include_dir('SDL3')]
```

Comment:
@icculus should we inject this include path like this?  Or are all uses expected to include via "SDL3/xxxx.h"?

Reply:
For SDL3, all users are supposed to include the SDL3 prefix:
```c
#include <SDL3/SDL.h>
```

Reply:
When you say "not supposed to" does that mean that `include/SDL3` is not on the default include path and `#include <SDL.h>` will not work?  e.g. in the cmake config for SDL3?   Or is it just a recommendation not to rely on that include path?   I feel like emscripten should do that same thing that the cmake or pkg-config does.. whatever that is.

Reply:
That's correct, `#include <SDL.h>` will not work.
https://github.com/libsdl-org/SDL/blob/9bd6d36471784e31109eab554d39e2aaf93cd56f/docs/README-migration.md?plain=1#L46

Reply:
Nice!  Thats good to know.  I will remove this.